flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1433,7 +1435,10 @@ object ScalarOperatorGens {
       .map(_._2.last)
       .values
       .toSeq
-    val valueType = mapType.getValueType
+    val valueType = resultType match {
+      case mapType1: MapType => mapType1.getValueType
+      case _ => DataTypes.INT().getLogicalType
```

Comment:
nit: it's better to call `new IntType()` at code gen level, I guess it should be not null?

Reply:
yep, you're right